decoder/req_decoder.go

```diff
@@ -65,23 +65,23 @@ func (mr monitoringReader) Close() error {
 
 func DecodeLimitJSONData(maxSize int64) ReqDecoder {
 	return func(req *http.Request) (map[string]interface{}, error) {
-		reader, err := readRequestJSONData(maxSize)(req)
+		contentType := req.Header.Get("Content-Type")
+		if !strings.Contains(contentType, "application/json") {
+			return nil, fmt.Errorf("invalid content type: %s", req.Header.Get("Content-Type"))
+		}
+
+		reader, err := CompressedRequestReader(maxSize)(req)
 		if err != nil {
 			return nil, err
 		}
 		return DecodeJSONData(monitoringReader{reader})
 	}
 }
 
-// readRequestJSONData makes a function that uses information from an http request to construct a Limited ReadCloser
-// of json data from the body of the request
-func readRequestJSONData(maxSize int64) Reader {
+// CompressedRequestReader makes a function that uses information from an http request to construct a Limited ReadCloser
+// from the body of the request, handling any decompression necessary
+func CompressedRequestReader(maxSize int64) ReqReader {
 	return func(req *http.Request) (io.ReadCloser, error) {
-		contentType := req.Header.Get("Content-Type")
-		if !strings.Contains(contentType, "application/json") {
-			return nil, fmt.Errorf("invalid content type: %s", req.Header.Get("Content-Type"))
-		}
```

Comment:
I don't think you should remove this check for v1

Reply:
The check has been moved to here and it's still in effect for v1: https://github.com/elastic/apm-server/pull/1260/files/fb77d985e9ee8623290bc23839542e9b318472c8#diff-5a9d3e83c01f2b2c9aa1bea1afade025R68

Reply:
Thanks @ron - not sure how I could overlook that.